src/components/folder/LinkList.js

```diff
@@ -1,66 +1,135 @@
-import { formatDate, getDaysAgo } from "../../util/DateCalculator";
+import { formatDate, getDaysAgo } from "../../util/date-calculator";
 import card from "../../images/linkbrary-card.png";
 import star from "../../images/star.png";
 import kebab from "../../images/kebab.png";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import AddModal from "../modals/AddModal";
+import DeleteModal from "../modals/DeleteModal";
 
-function LinkListItem({ linkData }) {
+function LinkListItem({ folderListData, linkData }) {
   const [kebabOn, setKebabOn] = useState(false);
+  const [addModalOn, setAddModalOn] = useState(false);
+  const [deleteModalOn, setDeleteModalOn] = useState(false);
   const url = linkData.url;
-  let image = linkData.image_source;
-  if (!image) {
-    image = card;
-  }
-  const handleKebabIn = (e) => {
+  let image = linkData.image_source || card;
+
+  const handleKebab = (e, isIn) => {
     e.preventDefault();
-    setKebabOn(true);
-  };
-  const handleKebabOut = () => {
-    setKebabOn(false);
+    setKebabOn(isIn);
   };
+
+  function handleAddModalOn() {
+    setAddModalOn(true);
+  }
+  function handleAddModalOff() {
+    setAddModalOn(false);
+  }
```

Comment:
함수를 하나로 합치면 더 간단하게 관리할 수 있을 것 같네요! handleKebab 처럼요!

```suggestion
  function handleAddModal(isOn) {
    setAddModalOn(isOn);
  }
```